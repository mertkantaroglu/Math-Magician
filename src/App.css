.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

.calculator {
  display: flex;
  align-items: center;
  flex-direction: row;
  margin: 10rem auto auto auto;
  flex-wrap: wrap;
  width: 300px;
}

.result {
  display: flex;
  color: #fff;
  justify-content: flex-end;
  background-color: rgba(150, 150, 150);
  width: 300px;
  height: 50px;
}

.result p {
  padding-right: 5px;
}

button {
  width: 75px;
  height: 50px;
  border: 1px solid lightgrey;
}

.operator {
  background-color: #ff8000;
}

.btn-zero {
  width: 150px;
}
