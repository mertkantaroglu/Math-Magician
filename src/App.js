import './App.css';
import { Routes, Route, Link } from 'react-router-dom';
import CalculatorPage from './pages/CalculatorPage';
import Home from './pages/Home';
import Quote from './pages/Quote';

function App() {
  return (
    <>
      <div className="app">
        <nav className="navbar">
          <h1>Math Magicians</h1>
          <ul className="navContainer">
            <li>
              <Link to="/">Home</Link>
            </li>
            <span>|</span>
            <li>
              <Link to="/calculatorpage">Calculator</Link>
            </li>
            <span>|</span>
            <li>
              <Link to="/quote">Quote</Link>
            </li>
          </ul>
        </nav>
      </div>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/calculatorpage" element={<CalculatorPage />} />
        <Route path="/quote" element={<Quote />} />
        <Route path="*" element={<h1>Page Not Found!</h1>} />
      </Routes>
    </>
  );
}

export default App;
