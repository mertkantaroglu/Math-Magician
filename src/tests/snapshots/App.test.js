import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from '../../App';

describe('App', () => {
  test('should match with App snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
