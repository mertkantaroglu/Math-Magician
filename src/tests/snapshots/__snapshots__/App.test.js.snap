// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match with App snapshot 1`] = `
Array [
  <div
    className="app"
  >
    <nav
      className="navbar"
    >
      <h1>
        Math Magicians
      </h1>
      <ul
        className="navContainer"
      >
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <span>
          |
        </span>
        <li>
          <a
            href="/calculatorpage"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <span>
          |
        </span>
        <li>
          <a
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  <div>
    <section
      className="home"
    >
      <h2>
        Welcome to our page!
      </h2>
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Eius ratione ex tempore error veniam voluptas consectetur ullam iste. Voluptatem accusamus harum numquam atque consectetur saepe architecto libero cum, totam dicta? Iste aspernatur rem quia. Voluptate iure magni minus animi assumenda. Ut laborum placeat nobis consequuntur qui ea ab assumenda voluptatem!Lorem ipsum dolor sit amet consectetur adipisicing elit. Quae qui ipsum magni impedit enim. Officiis iste saepe, ipsam minus veritatis architecto?
      </p>
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Eius ratione ex tempore error veniam voluptas consectetur ullam iste. Voluptatem accusamus harum numquam atque consectetur saepe architecto libero cum, totam dicta? Iste aspernatur rem quia. Voluptate iure magni minus animi assumenda. Ut laborum placeat nobis consequuntur qui ea ab assumenda voluptatem!Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </section>
  </div>,
]
`;
